src/nvidia.cc

```diff
@@ -309,10 +334,35 @@ struct nvidia_s {
 	ATTR_ID attribute;
 	char *token;
 	SEARCH_ID search;
+//  added new field for GPU id
+    int gpu_id;
+};
+
+//Cache by value
+struct nvidia_c_value {
+    int memtotal = -1;
+	int gputempthreshold = -1;
+};
```

Comment:
Minor formatting spot: it looks like you've mixed tabs and spaces here. Probably best to do a quick find/replace (I think the file was tab-indented before)